linux/git_mirror

```diff
@@ -813,21 +848,181 @@ function git_push_main()
           # should not have any remote refs. In any case, these cause an error,
           # "deny updating a hidden ref:", when pushing to a gitlab server. For now,
           # only filter out refs/remotes
-          #
-          # Push to the mirror (git push --mirror will remove deleted refs)
+          # RE It is possible to get refs/remotes/ if a bare repo fetches from
+          # another (bare) repo, which could then be pushed to, e.g., github
+
           # Spaces are not allowed in a refname
-          "${GIT}" for-each-ref --format='%(refname)' | grep -v -e '^refs/remotes' -e '^refs/stash' | \
-              xargs "${GIT}" push "${repo_urls[$index]}"
+          local refs=($("${GIT}" for-each-ref --format='%(refname)' | \
+              grep -E "${GIT_MIRROR_PUSH_REFS_REGEX}" | \
+              grep -v -E "${GIT_MIRROR_SKIP_REFS_REGEX}" | \
+              grep -v -e '^refs/remotes' -e '^refs/stash' || :))
+          # xargs tries all args before returning an error code
+          # per xarg's man page:
+          # "The xargs utility exits immediately (without processing any
+          # further input) if ... an invocation of utility exits with a value
+          # of 255", which is not the case if pushing a branch fails
+          # (If refs is empty, the echo succeeds but there is nothing to pass
+          # through the pipe, so the command short-circuits)
```

Comment:
If you want to use that exit 255 behavior, you could always:

```
echo blah | xargs bash -c '"${GIT}" push "${repo_url}" || exit 255'
```

Reply:
I actually do not want that behavior